src/urls.ts

```diff
@@ -22,27 +22,49 @@ export const getWorkUrl = ({
   return workUrl;
 };
 
-export const getAsShortUrl = ({ url }: { url: string }) => url.replace(/archiveofourown/, 'ao3');
+export const getAsShortUrl = ({ url }: { url: string }) =>
+  url.replace(/archiveofourown/, "ao3");
 
 export const getUserProfileUrl = ({ username }: { username: string }) =>
   `https://archiveofourown.org/users/${encodeURI(username)}/profile`;
 
-const tokenReplacerObject = {
-  '/': '*s*',
-  '&': '*a*',
-  '.': '*d*',
-  '#': '*h*',
-  '?': '*q*'
-}
-const tokensToEscape = ['/', '?', '.']
-const replaceEscapableTokens = (c:string) => tokensToEscape.includes(c) ? `\\${c}` : c;
-const tagUrlReplaceChars = new RegExp(`(${Object.keys(tokenReplacerObject).map(replaceEscapableTokens).join('|')})`, 'g')///(\/|\.|&|#|\?)/g;
+const TOKEN_REPLACEMENTS_MAP = {
+  "/": "*s*",
+  "&": "*a*",
+  ".": "*d*",
+  "#": "*h*",
+  "?": "*q*",
+} as const;
 
-type ReplacerObjectKeys = keyof typeof tokenReplacerObject
+type ReplaceableToken = keyof typeof TOKEN_REPLACEMENTS_MAP;
+
+const REPLACEABLE_TOKENS = Object.keys(
+  TOKEN_REPLACEMENTS_MAP
+) as ReplaceableToken[];
+
+const TOKENS_TO_ESCAPE = ["/", "?", "."];
+
+const shouldEscapeToken = (c: string) => TOKENS_TO_ESCAPE.includes(c);
+const isReplaceableToken = (c: string): c is ReplaceableToken =>
```

Comment:
How I broke these down is a bit of a preference, but I felt this split was a bit more readable. 

TBH looking at this again I'd name `isReplaceableToken` as `shouldReplaceToken`.